src/scripts/main.js

```diff
@@ -1,3 +1,31 @@
 'use strict';
 
-// write code here
+const headlines = document.querySelector('thead').rows[0];
+const list = document.querySelector('tbody');
+const listRows = list.rows;
+
+headlines.addEventListener('click', (e) => {
+  const headline = e.target.closest('th');
+
+  const index = headline.cellIndex;
+
+  [...listRows].sort((a, b) => {
+    const row = a.cells[index].textContent;
+    const nextRow = b.cells[index].textContent;
+
+    if (/\d/.test(row)) {
+      const checkedRow = convertToNumber(row);
+      const checkedNextRow = convertToNumber(nextRow);
+
+      return Number(checkedRow) - Number(checkedNextRow);
+    }
+
+    return row.localeCompare(nextRow);
+  }).forEach(row => list.appendChild(row));
+});
+
+function convertToNumber(string) {
+  const number = Number(string.replace(/[$,]/g, ''));
+
+  return `${number}`;
```

Comment:
The convertToNumber function should return a number, not a stringified number. Remove the template literal and return the number directly.